src/Views/Components/ListTable/ListTable.tsx

```diff
@@ -0,0 +1,248 @@
+import {useEffect, useState} from 'react';
+import {__, _n, sprintf} from '@wordpress/i18n';
+import cx from 'classnames';
+
+import styles from './ListTable.module.scss';
+import ListTableRows from './ListTableRows';
+import {Spinner} from '../index';
+import {forEach} from "lodash";
+
+export interface ListTableProps {
+    //required
+    columns: Array<ListTableColumn>;
+    title: string;
+    data: {items: Array<{}>};
+
+    //optional
+    pluralName?: string;
+    singleName?: string;
+    rowActions?: (({item, data, addRow, removeRow}) => JSX.Element)|JSX.Element|JSX.Element[]|Function|null;
+    parameters?: {};
+    error?: {}|Boolean;
+    isLoading?: Boolean;
+    align?: 'start'|'center'|'end';
+}
+
+export interface ListTableColumn {
+    name: string;
+    text: string;
+    inlineSize?: string;
+    preset?: string;
+    heading?: boolean;
+    alignColumn?: 'start'|'center'|'end';
+    addClass?: string;
+    render?: ((item: {}) => JSX.Element)|JSX.Element|JSX.Element[]|null;
+}
+
+const toggleAllRowCheckboxes = (event) => {
+    const checked = event.target.checked;
+    const checkboxes = document.querySelectorAll('input.giveListTableSelect');
+    checkboxes.forEach((checkbox: HTMLInputElement) => {
```

Comment:
I think this particular situation is a trade-off - uncontrolled inputs offer better performance and avoid certain React pitfalls like losing focus when an input re-renders. (`react-hook-form` is a good example of this, and you can see their rationale [here](https://react-hook-form.com/faqs).)

In this case, since the checkboxes for individual rows are far down the component tree, they would need to share their state with `ListTablePage`, `ListTable`, and `ListTableRows`. This would end up causing extra re-renders for the whole page every time a checkbox got checked or unchecked, regardless of whether we use context or a reducer. And adding all the internal logic to manage that state will make the components more fragile.

But we only need to programmatically modify the checkboxes when someone clicks the 'Select All' checkbox (the one in the top left corner of the table). And we only need to read their value when a user clicks the 'Apply' button for a bulk action. So DOM manipulation (here, at least) offers us a more performant way to achieve the same result.